pkg/controller/dgraph/models/query/cluster.go

```diff
@@ -22,27 +22,26 @@ import (
 	"github.com/vmware/purser/pkg/controller/dgraph"
 )
 
+var executeQuery = dgraph.ExecuteQuery
+var executeQueryRaw = dgraph.ExecuteQueryRaw
+
+func getClusterHierarchyQuery(view string) string {
+	switch view {
+	case Physical:
+		return getHierarchyQueryForPhysicalResource()
+	case Logical:
+		return getHierarchyQueryForLogicalResource()
+	default:
+		return ""
+	}
+}
+
 // RetrieveClusterHierarchy returns all namespaces if view is logical and returns all nodes with disks if view is physical
 func RetrieveClusterHierarchy(view string) JSONDataWrapper {
-	var query string
-	if view == Physical {
-		query = `query {
-			children(func: has(name)) @filter(has(isNode) OR has(isPersistentVolume)) {
-				name
-				type
-			}
-		}`
-	} else {
-		query = `query {
-			children(func: has(isNamespace)) {
-				name
-				type
-			}
-		}`
-	}
+	query := getClusterHierarchyQuery(view)
 
 	parentRoot := ParentWrapper{}
-	err := dgraph.ExecuteQuery(query, &parentRoot)
+	err := executeQuery(query, &parentRoot)
```

Comment:
I'll suggest a rename here. 
Make the `getClusterHierarchyQuery(view string)` a public function as `RetrieveClusterHierarchy(view string)` and the other function as private as `executeQuery(view string) JSONDataWrapper` making the purpose of the functions more clear.

Reply:
`executeQuery` is a variable set equal to `dgraph.ExecuteQuery`. This will allow us to mock dgraph execute query simply by changing `executeQuery` to a mock-dgraph-query-execution